### DISCUSSION TITLE
"ALL_NEW" property inside return parameter for Model Update

### DISCUSSION THREAD

**Discussion by Indridcold96** [sensedeep/dynamodb-onetable#315] [General]
I have an issue ragarding the return parameter when using the update method. We are using Onetable version 2.0.2, our goal is to retrieva all the fields even when we send a light payload. I have a defined model with several fields, not all are required, the desired behavior is, even if a light payload is provided, the system should return all the values from the fields defined in the schema, however, that is not happening. To explain the issue, below is a simple example:

```
	const client = new Dynamo({
		client: new DynamoDBClient({
			region: "foo1",
			requestHandler: new NodeHttpHandler({
				connectionTimeout: 600,
				socketTimeout: 600
			}),
			maxAttempts: 3
		})
	});

	class ModelNames {
		public static readonly Foo = "Foo";
	}

	const fooBarBazSchema = {
		version: '1.0.0',
		indexes: {
			primary: { hash: 'pk', sort: 'sk' },
			gs1: { hash: 'gs1pk', sort: 'gs1sk' }
		},
		models: {
			Foo: {
				pk: { type: String, value: 'foo#${key}' },
				sk: { type: String, value: 'foo#' },

				gs1pk: { type: String, value: '${_type}' },
				gs1sk: { type: String, value: 'foo#${isFoo}' },

				field1: { type: String, required: true, unique: true },
				field2: { type: String, required: true },
				field3: { type: String, default: null }
			}
	};

	export const table = new Table({
		client: client,
		name: 'FooBarBazdb',
		schema: fooBarBazSchema,
	});
	
	export { ModelNames };
	export type FooType = Entity<typeof fooBarBazSchema.models.Foo>
```

Then, we define an interface to wrap the values:

```
	interface IFoo {
		field1: string,
		field2: string,
		field3?: String
	}

	export { IFoo };
```

We then have the following class:

```
	class FooAccessor implements IFooAccessor {
		constructor() {
			this._model = table.getModel<FooType>(ModelNames.Foo);
		}

		_model: Model<FooType>;

		async updateFoo(updatedFoo: IFoo): Promise<IFoo> {
			const result = await this._model.update({...updatedFoo}, {return: "ALL_NEW"});
			return result as IFoo;
		}
	}
```
	
When an update request is made using the accessor, even though we added the "ALL_NEW" property to the return parameter, it only returns the fields sent with the payload. For example we send a request with the following payload:

```
{
    "data": {
        "field1": "b50da647-1fbc-4aad-aa3a-c1da42f7a9e8",
        "field2": "UpdateFoo",
    }
}
```

We expect it to return:

```
{
    "data": {
        "field1": "b50da647-1fbc-4aad-aa3a-c1da42f7a9e8",
        "field2": "UpdateFoo",
		"field3": "StaticFoo",
    }
}

```
However, it only returns this:

```
{
    "data": {
        "field1": "b50da647-1fbc-4aad-aa3a-c1da42f7a9e8",
        "field2": "UpdateFoo",
    }
}
```

Because of the addition of the "ALL_NEW" parameter, we thought it should return all fields. Is there a way we can retrieve all the values from all the fields even when we are sending a light payload without having to do a fetch after the update?

Also, in the future it would be nice to see this case as an example in the documentation, because we are not sure we are doing it right.

**Comment by mobsense** [comment]
You don't need to use {return}. By default, update will return all values.

Use log: true to see what the actual command is:

```
const result = await model.update({...updatedFoo}, {log: true})
```

**Reply by Indridcold96** [reply]
Will try the log parameter to see what it is being passed to DynamoDB, but from what I've checked, the version of Onetable I'm using is 2.0.2, and there was a fix to an issue, fix #245, that should deal with our case, is that correct?

**Reply by mobsense** [reply]
Yes, from memory that was a short regression that was fixed. Recommend upgrading.
